clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/RestoreSnapshotAtomicOperation.groovy

```diff
@@ -104,43 +139,86 @@ class RestoreSnapshotAtomicOperation implements AtomicOperation<Void> {
       }
     }
 
+    task.updateStatus BASE_PHASE, "Restoring snapshot with timestamp ${this.snapshotTimestamp} for application ${this.applicationName} in account ${this.accountName}"
+    createTerraformConfig()
+    ArrayList<String> command = ["terraform", "apply", "-state=$directory/terraform.tfstate", "$directory"]
+    String jobId = jobExecutor.startJob(new JobRequest(tokenizedCommand: command), System.getenv(), new ByteArrayInputStream())
+    waitForJobCompletion(jobId)
+
     cleanUpDirectory()
     return null
   }
 
   private Void importServerGroupState(GoogleServerGroup.View serverGroup) {
-    importResource("google_compute_instance_group_manager", serverGroup.name)
+    importResource("google_compute_instance_group_manager", serverGroup.name, serverGroup.name, serverGroup.region)
     def instanceTemplate = serverGroup.imageSummary.image
-    if (instanceTemplate) {
-      importResource("google_compute_instance_template", serverGroup.imageSummary.image.name)
+    if (instanceTemplate && !imported.contains(instanceTemplate.name)) {
+      importResource("google_compute_instance_template", instanceTemplate.name, instanceTemplate.name, serverGroup.region)
+      imported.add(instanceTemplate.name)
       if (instanceTemplate.properties.tags?.items) {
         applicationTags.addAll(instanceTemplate.properties.tags.items)
       }
     }
     if (serverGroup.autoscalingPolicy) {
-      importResource("google_compute_autoscaler", serverGroup.name)
+      importResource("google_compute_autoscaler", serverGroup.name, serverGroup.name, serverGroup.region)
+    }
+    // Add dependencies of existing resources to the state file
+    def stateFile = new File("$directory/terraform.tfstate")
+    String stateJson = new Scanner(stateFile).useDelimiter("\\Z").next()
+    Map state = objectMapper.readValue(stateJson, new TypeReference<HashMap<String,Object>>() {})
+    def igmDependencies = serverGroup.loadBalancers.collect { String loadBalancer ->
+      return "google_compute_target_pool.$loadBalancer"
+    }
+    igmDependencies << "google_compute_instance_template.$instanceTemplate.name"
+    addDependencies(state.modules[0].resources["google_compute_instance_group_manager.$serverGroup.name"], igmDependencies)
```

Comment:
modules should never be empty here because I imported an instance group successfully, I have tested in the case where there is no infrastructure and it still works